lib/puppet/parser/functions/fail_is4x.rb

```diff
@@ -0,0 +1,3 @@
+Puppet::Parser::Functions::newfunction(:fail_is4x, :arity => 1, :doc => 'Fail with a parse error indicating that 4x function API should be used.') do |args|
+    raise Puppet::ParseError, "#{args[0]}() can only be called using the 4.x function API. See Scope#call_function"
+end
```

Comment:
Not a very good idea (sorry) - this function ends up as something users can call, it shows up in documentation and what not. Remove this.
If you feel that it is bad to copy/paste the error message, create an error class instead that takes the function name as argument.